openslides_backend/action/actions/meeting/archive.py

```diff
@@ -22,6 +23,41 @@ class MeetingArchive(UpdateAction, GetMeetingIdFromIdMixin):
     model = Meeting()
     schema = DefaultSchema(Meeting()).get_update_schema()
 
+    def validate_instance(self, instance: dict[str, Any]) -> None:
+        super().validate_instance(instance)
+
+        messages = []
+
+        active_speaker_exists = self.datastore.exists(
+            "speaker",
+            And(
+                FilterOperator("meeting_id", "=", instance["id"]),
+                FilterOperator("begin_time", "!=", None),
+                FilterOperator("end_time", "=", None),
+            ),
+            lock_result=False,
+        )
+        active_polls = self.datastore.filter(
+            "poll",
+            And(
+                FilterOperator("meeting_id", "=", instance["id"]),
+                FilterOperator("state", "=", "started"),
+            ),
+            ["title"],
+            lock_result=False,
+        )
+
+        if active_speaker_exists:
+            messages.append("speaker")
+        if len(active_polls):
+            messages.append(
+                f"polls ({', '.join([model['title'] for model in active_polls.values()])})"
+            )
+        if len(messages):
+            raise ActionException(
+                f"Cannot archieve meeting with active {' and '.join(messages)}."
+            )
```

Comment:
`validate_instance` is for payload format checking and should not include any datastore calls, please move this to `update_instance`.